### PR TITLE
Add Expander by References 

### PR DESCRIPTION
**Pull request by ratuspro**

### CODE REVIEW

File: sciencer/expanders/expand_by_references.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        Expander (_type_): _description_
+    """
+
+    def __init__(self) -> None:
+        super().__init__([])
+
+    def execute(self, papers: List[Paper], providers: List[Provider]) -> List[Paper]:
+        resulting_papers = set()
+
+        referenced_papers_ids = set()
```
Comment (hineios) [inline sciencer/expanders/expand_by_references.py:22]: Just merge this `for` into the next. Unless you have an hidden agenda going on here, this just seems like a waste of space...

**Review by hineios — CHANGES_REQUESTED** [review]
Could be interesting to consider an optional argument for specify how many nested levels of references to expand.

But in a next iteration maybe.

File: sciencer/expanders/expand_by_references.py (hunk 1 above)
Comment (ratuspro) [reply sciencer/expanders/expand_by_references.py:22]: The first for is used to identify all the referenced papers (without duplicates)
The second for is used to expand these referenced papers (avoiding redundant expansions)

Comment (hineios) [reply sciencer/expanders/expand_by_references.py:22]: Ah. The duplicates part of the first for had eluded me.

**Pull request merged**